api/proto/auth/config.go

```
@@ -0,0 +1,97 @@
+package auth
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"time"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/trace"
+)
+
+// Config contains configuration of the client
+type Config struct {
+	// Addrs is a list of addresses to dial
+	Addrs []string
+	// Dialer is a custom dialer, if provided
+	// is used instead of the list of addresses
+	Dialer ContextDialer
+	// KeepAlivePeriod defines period between keep alives
+	KeepAlivePeriod time.Duration
+	// KeepAliveCount specifies amount of missed keep alives
+	// to wait for until declaring connection as broken
+	KeepAliveCount int
+	// TLS is a TLS config
+	TLS *tls.Config
+}
+
+// CheckAndSetDefaults checks and sets default config values
+func (c *Config) CheckAndSetDefaults() error {
+	if len(c.Addrs) == 0 && c.Dialer == nil {
```

If both are set by the user, do we also want to return an error?